src/Core/src/Hosting/MauiAppBuilder.cs

```diff
@@ -74,91 +56,57 @@ internal MauiAppBuilder(bool useDefaults = true)
 		/// <summary>
 		/// A collection of logging providers for the application to compose. This is useful for adding new logging providers.
 		/// </summary>
-		public ILoggingBuilder Logging => _logging;
+		public ILoggingBuilder Logging { get; }
 
 		/// <summary>
-		/// An <see cref="IHostBuilder"/> for configuring host specific properties, but not building.
-		/// To build after configuration, call <see cref="Build"/>.
+		/// Registers a <see cref="IServiceProviderFactory{TBuilder}" /> instance to be used to create the <see cref="IServiceProvider" />.
 		/// </summary>
-		public IHostBuilder Host => _host;
+		/// <param name="factory">The <see cref="IServiceProviderFactory{TBuilder}" />.</param>
+		/// <param name="configure">
+		/// A delegate used to configure the <typeparamref T="TBuilder" />. This can be used to configure services using
+		/// APIS specific to the <see cref="IServiceProviderFactory{TBuilder}" /> implementation.
+		/// </param>
+		/// <typeparam name="TBuilder">The type of builder provided by the <see cref="IServiceProviderFactory{TBuilder}" />.</typeparam>
+		/// <remarks>
+		/// <para>
+		/// <see cref="ConfigureContainer{TBuilder}(IServiceProviderFactory{TBuilder}, Action{TBuilder})"/> is called by <see cref="Build"/>
+		/// and so the delegate provided by <paramref name="configure"/> will run after all other services have been registered.
+		/// </para>
+		/// <para>
+		/// Multiple calls to <see cref="ConfigureContainer{TBuilder}(IServiceProviderFactory{TBuilder}, Action{TBuilder})"/> will replace
+		/// the previously stored <paramref name="factory"/> and <paramref name="configure"/> delegate.
+		/// </para>
+		/// </remarks>
+		public void ConfigureContainer<TBuilder>(IServiceProviderFactory<TBuilder> factory, Action<TBuilder>? configure = null) where TBuilder : notnull
```

Comment:
How many of the 3rd party containers expose their IServiceProviderFactory\<T\>?

Reply:
Public:

https://github.com/dadhi/DryIoc/blob/c24353943ed9fae30ef097ea2e30c0ae6bc6f43e/src/DryIoc.Microsoft.DependencyInjection/DryIocAdapter.cs#L62

https://github.com/JasperFx/lamar/blob/0bc88c424430ec13a0b17699e53ccbc5f8e855e0/src/Lamar.Microsoft.DependencyInjection/LamarServiceProviderFactory.cs#L6

https://github.com/seesharper/LightInject.Microsoft.DependencyInjection/blob/c173fc343a2272e60cfd8c2a473b81d77570272e/src/LightInject.Microsoft.DependencyInjection/LightInject.Microsoft.DependencyInjection.cs#L264

https://github.com/z4kn4fein/stashbox-extensions-dependencyinjection/blob/098de1aef9afeb54a1a0d81871975aad8aeddec3/src/stashbox.extensions.dependencyinjection/StashboxServiceProviderFactory.cs#L9

https://github.com/unitycontainer/microsoft-dependency-injection/blob/0a428ed1b8aadcccef056f8b437c88778fe99bad/src/ServiceProvider/ServiceProviderFactory.cs#L7

https://github.com/autofac/Autofac.Extensions.DependencyInjection/blob/304bf20a50e4ae8f9ef8edf91bedc033ea4ac303/src/Autofac.Extensions.DependencyInjection/AutofacServiceProviderFactory.cs#L35

Non-public:

https://github.com/ipjohnson/Grace.DependencyInjection.Extensions/blob/425c2748f048c5eea1676df65040782b78f30dc2/src/Grace.AspNetCore.Hosting/GraceServiceProviderExtensions.cs#L42

Note that Grace will have the same problem in a Blazor WASM app (I took this same design from [WebAssemblyAppBuilder](https://github.com/dotnet/aspnetcore/blob/f9de7e4b9d4816c26db0652c4b7c99f321a16b44/src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs#L216)). 

I've opened https://github.com/ipjohnson/Grace.DependencyInjection.Extensions/issues/32 on Grace to consider this scenario.

Reply:
FYI `ipjohnson` already fixed https://github.com/ipjohnson/Grace.DependencyInjection.Extensions/issues/32. So now all the 3rd party containers expose their IServiceProviderFactory<T>.